src/Analyzers/Analyzers/test/CSharpCodeFixVerifier`2+Test.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    public static partial class CSharpCodeFixVerifier<TAnalyzer, TCodeFix>
+        where TAnalyzer : DiagnosticAnalyzer, new()
+        where TCodeFix : CodeFixProvider, new()
+    {
+        public class Test : CSharpCodeFixTest<TAnalyzer, TCodeFix, XUnitVerifier>
+        {
+            public Test()
+            {
+                ReferenceAssemblies = CodeAnalysis.Testing.ReferenceAssemblies.NetCore.NetCoreApp50;
+
+                foreach (var assembly in Directory.EnumerateFiles(AppContext.BaseDirectory, "*.dll"))
+                {
+                    TestState.AdditionalReferences.Add(CreateReferenceFromFile(assembly));
+                }
```

Comment:
📝 This is carried over from the old tests, but is not the preferred way to test analyzers. Ideally, this is converted from loose file references to proper NuGet package references, ensuring that the analyzers will work as expected when downstream consumers bring them in along with other NuGet references.

Reply:
>  Ideally, this is converted from loose file references to proper NuGet package references, ensuring that the analyzers will work as expected when downstream consumers bring them in along with other NuGet references.

Can you clarify with some context what `this` and `bring them in along with other NuGet references` mean?

One thing that might not be obvious is that our analyzers ship in the .NET SDK, and *most* of what we build does not ship as packages.